lib/ramble/ramble/language/modifier_language.py

```diff
@@ -35,7 +35,7 @@ def _execute_mode(mod):
     return _execute_mode
 
 
-@modifier_directive('default_modes')
+@modifier_directive(dicts=())
```

Comment:
This is a pattern we borrowed from spack. Other directives that are similar are `tags` and `maintainers`.

But also, the `default_modes` name in the original directive was not used. The directive set `mod.default_mode` and ignored the dict `mod.default_modes` entirely. So, this change removes a dictionary from being created by the `default_mode` directive (since we really don't want a dict anyway).

Reply:
Yea I was having a hard time coming up with a nice way. It felt a bit like we might be able to check if the class it wants to meta into has already been instantiated but I wasn't quite sure how that would look 

Happy either way but maybe at the very least we need a comment wanring people they will have a bad time if they call this function at non-init time?

Reply:
I actually changed the internal attribute named to `_default_usage_mode` to make this method callable at init time. So, it should work. But still, the attribute we want is a scalar string, not a dict (which is why you see the `dicts=()` bit in the directive decorator).